cmd/argocd/commands/app.go

```diff
@@ -1621,6 +1637,24 @@ func NewApplicationSyncCommand(clientOpts *argocdclient.ClientOptions) *cobra.Co
 	return command
 }
 
+func getAppNamesBySelector(ctx context.Context, appIf applicationpkg.ApplicationServiceClient, selector string) ([]string, error) {
+	appNames := []string{}
+	if selector != "" {
+		list, err := appIf.List(ctx, &applicationpkg.ApplicationQuery{Selector: pointer.String(selector)})
+		if err != nil {
+			return []string{}, nil
+		}
+		// unlike list, we'd want to fail if nothing was found
+		if len(list.Items) == 0 {
+			return []string{}, fmt.Errorf("no apps match selector %v", selector)
+		}
+		for _, i := range list.Items {
+			appNames = append(appNames, i.Name)
+		}
+	}
+	return appNames, nil
+}
+
```

Comment:
@danielhelfand how would you feel about pushing this logic server-side to the Delete handler in server/application.go? This would more nicely match what the List handler does.

Would require adding a `projects` field to the delete request object in application.prot, running codegen, and modifying the Delete handler in application.go.

Reply:
I looked at this some more. It wouldn't make sense to do this server-side, because that would prevent the per-app delete confirmation from working as expected.

Reply:
Thanks for looking over and sorry for hiatus on this.